depends/libhdfs3/src/client/Hdfs.cpp

```diff
@@ -35,6 +35,7 @@
 #include "Thread.h"
 #include "XmlConfig.h"
 
+#include <boost/scope_exit.hpp>
```

Comment:
Do not make boost as hard requirement. libhdfs3 only needs boost if you are using an old version compiler such as GCC 4.4.

If the function provided by boost is not in the standard c++ library, do not use such function.

Reply:
Removed. Which C++ version do you target for libhdfs3? 

Also, just out of curiosity where does the compiler version constraint come from? CentOS 6?

Reply:
I did not means the compiler version constrain, just use c++ standard library first, and use boost as an alternative.